s3/src/main/scala/geotrellis/spark/io/s3/package.scala

```diff
@@ -26,4 +31,27 @@ package object s3 {
       .mkString("/")
 
   implicit class withSaveToS3Methods[K](rdd: RDD[(K, Array[Byte])]) extends SaveToS3Methods(rdd)
+
+  def s3ObjectExists(bucket: String, key: String, s3Client: S3Client): Boolean = {
```

Comment:
These should be extension methods on the AWS S3 client.
Also, it's possible and even common that `HEAD` request will not be available on a bucket. I've seen this happen with requester pays buckets.

Reply:
Looks like the `ListBucket` perms are sufficient to infer a failed `HEAD` request's cause: https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectHEAD.html